### PR TITLE
[CARBONDATA-1433] Added Vectorized Reader for Presto Integration

### PR DESCRIPTION
**Pull request by bhavya411**
This PR is for optimizing the Presto Integration Performance
- Added Vectorized Reader for reading the data
- Used DictionaryBlock for loading the dictionary values.
- Removed unused code 

- Build is Successful
- Verified that all test cases are passing
- Manually Verified by running queries

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/356/

**Comment by chenliang613** [comment]
retest this please

**Comment by chenliang613** [comment]
retest this please

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/505/

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbonTypeUtil.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package org.apache.carbondata.presto;
+
+import org.apache.carbondata.core.metadata.datatype.DataType;
+
+import org.apache.spark.sql.types.DataTypes;
```
Comment (jackylk) [inline integration/presto/src/main/java/org/apache/carbondata/presto/CarbonTypeUtil.java]: should not depend on spark in presto integration module

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbonVectorizedRecordReader.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+import org.apache.spark.sql.execution.vectorized.ColumnarBatch;
+import org.apache.spark.sql.types.DecimalType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A specialized RecordReader that reads into InternalRows or ColumnarBatches directly using the
+ * carbondata column APIs and fills the data directly into columns.
+ */
+class CarbonVectorizedRecordReader extends AbstractRecordReader<Object> {
```
Comment (jackylk) [inline integration/presto/src/main/java/org/apache/carbondata/presto/CarbonVectorizedRecordReader.java]: Should not copy code from spark2 integration module. I think the correct way is to move Vector reader outside of spark integration and share with all integration modules.

Comment (chenliang613) [reply integration/presto/src/main/java/org/apache/carbondata/presto/CarbonVectorizedRecordReader.java]: Yes, we could consider the refactor as you suggested in 1.3.0.  
@bhavya411  please add comment for this class , like "TODO: abstract and consolidate VectorizedRecordReader into one , and be shared for all integration modules"

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/593/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/598/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**